### ISSUE TITLE
Compiler crash when CRLF line-ending sequence is used

### ISSUE THREAD

**Issue by Namek** [mint-lang/mint#173]
I have 2 `.mint` source files. First one (Main) has this:

```
store Game {
  state moveHistory : Array(Vec2) = []
}
```

2nd one has this:
```
record Vec2 {
  x : Number
  y : Number
}
```

Note the lack of comma after `x : Number`. Then I can see the error below in the console. If the record definition is in same file as usage then I get a more meaningful error.

```
Mint - Running the development server
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
⚙ Ensuring dependencies... 5.501ms
Unhandled exception: Negative count: -18 (ArgumentError)
  from /usr/share/crystal/src/pointer.cr:0:11 in '[]'
  from /home/travis/build/mint-lang/mint/src/utils/html_snippet.cr:50:9 in 'render'
  from /home/travis/build/mint-lang/mint/src/message.cr:171:9 in 'to_html'
  from /home/travis/build/mint-lang/mint/src/reactor.cr:0:22 in 'parse_and_run'
  from /home/travis/build/mint-lang/mint/src/commands/command.cr:2:3 in 'parse_and_run'
  from /home/travis/build/mint-lang/mint/src/mint.cr:3:1 in '???'
  from /usr/share/crystal/src/crystal/main.cr:47:14 in 'main'
  from ???
  from ???
```

**Comment by Namek** [comment]
Sorry, that was the old version (0.5.0)

**Comment by Namek** [comment]
The above was in 0.5.0, the below is for 0.7.1:

```
Unhandled exception: Negative count: -17 (ArgumentError)
  from /usr/share/crystal/src/array.cr:0:0 in '[]'
  from /home/travis/build/mint-lang/mint/src/utils/html_snippet.cr:50:9 in 'render'
  from /home/travis/build/mint-lang/mint/src/message.cr:171:9 in 'to_html'
  from /home/travis/build/mint-lang/mint/src/reactor.cr:50:13 in 'parse_and_run'
  from /home/travis/build/mint-lang/mint/src/commands/command.cr:2:3 in 'parse_and_run'
  from /usr/share/crystal/src/crystal/main.cr:47:14 in 'main'
  from ???
  from ???
  from ???
```

**Comment by gdotdesign** [comment]
Thanks for reporting :+1:

**Comment by gdotdesign** [comment]
I've tried to reproduce this on `0.7.1` and `master` without success. I pushed it to this repo https://github.com/gdotdesign/bug-173-repro can you try it?

I assume it errors out when dev server is started? Which OS are you on?

**Comment by Namek** [comment]
I've found out this actually happens pretty often but I'm unable to find a small example to reproduce. The situations were like:
- using something that is defined below (when defined above then it worked)
- using a function by name which is not defined yet

I'll get back here when some SSSCE appears at me. But the hints may be:
- (maybe) more modules than one
- something is not yet defined/parsed so it can't find its name?

**Comment by Namek** [comment]
I have tried several things to find what's the cause but failed at it. It'd be probably easier debugging the compiler. Here's the repro, not SSSCE though: https://github.com/Namek/hashi-mint/commit/5c3e6e885a02b10de47562a021fe09c598601368

**Comment by s0kil** [comment]
@Namek Recursion is not supported right now.

**Comment by s0kil** [comment]
There seems to be some sort of encoding issue with the `Generation.mint` file, I tried changing the encoding to UTF8 and that stopped the compiler from throwing.
This could also be achieved by running `mint format source/Generation.mint`.

**Comment by Namek** [comment]
> @Namek Recursion is not supported right now.

The docs and changelog both say it is. And it seems to compile.

> There seems to be some sort of encoding issue with the `Generation.mint` file, I tried changing the encoding to UTF-8 and that stopped the compiler from throwing.
> This could also be achieved by running `mint format source/Generation.mint`.

@s0kil Thanks! It's not encoding but actually the line-ending sequence - Windows-y CRLF instead of Linux-y LF.

**Comment by s0kil** [comment]
`format=$(iconv -c -t UTF-8 source/Generation.mint) echo $format > source/Generation.mint`
`tail -c +4 source/Generation.mint`
Outputs:
```
}slandIdx2 : Number,ation,|> Array.slice(0, Array.size(leftBridges)), acc)(Island) {land.maxConnectionCounts, connectionSizesToAdd) } )}
```
That is the problem.

**Comment by s0kil** [comment]
Like you said, changing EOF Sequence to LF also works.

**Comment by gdotdesign** [comment]
I think it can be fixed here https://github.com/mint-lang/mint/blob/master/src/utils/html_snippet.cr#L12 but I'm not sure how do to it to support all line endings properly.

**Comment by gdotdesign** [comment]
It should be fixed with cc9117daa2009436a406953ae43744b50ce980f1. I could not test it properly, @Namek could you try?

**Comment by Namek** [comment]
I cannot reproduce the error anymore, seems to work, thanks!
